### PR TITLE
Use `-Xlinker` for `-install_name`

### PR DESCRIPTION
**Pull request by fmeum**
This supports `,` in `runtime_solib_name`.

### CODE REVIEW

**Comment by fmeum** [comment]
Would it be possible to get a release after this has been merged? This would unblock enabling C++ remote execution tests for macOS in Bazel (https://github.com/bazelbuild/bazel/pull/23085).

**Comment by fmeum** [comment]
Target names with commas can trigger this (there's a Bazel test for this case). The feature above this one already uses `-Xlinker`. If deduplication doesn't work, that sounds like a bug we should solve regardless.

**Review by keith — APPROVED** [review]

**Comment by keith** [comment]
fwiw we tried to fix that but weren't able to land a pr upstream

**Comment by fmeum** [comment]
Can you point me to the PR?

**Comment by keith** [comment]
see the deduplication stuff in this issue https://github.com/bazelbuild/bazel/issues/16939#issue-1480421627

**Pull request merged**
